File: Wki.EventSourcing/Messages/OfficeActorState.cs
using System;
using Wki.EventSourcing.Util;

namespace Wki.EventSourcing.Messages
{
    public class OfficeActorState
    {
        // start information
        public DateTime StartedAt { get; internal set; }

        // durable actors
        public int NrActorsLoaded { get; internal set; }
        public DateTime LastActorLoadedAt { get; internal set; }
        public int NrActorsRemoved { get; internal set; }
        public DateTime LastActorRemovedAt { get; internal set; }

        // message handling
        public int NrCommandsForwarded { get; internal set; }
        public int NrUnhandledMessages { get; internal set; }

        public OfficeActorState()
        {
            var now = SystemTime.Now;
            StartedAt = now;
            NrActorsLoaded = 0;
            NrActorsRemoved = 0;
            NrCommandsForwarded = 0;
            NrUnhandledMessages = 0;
        }
    }
}

File: Designer.Web/www/app/dto/officeactorstate.ts
import { OfficeActorChildState } from './officeactorchildstate';

class ActorInfo {
    constructor(
        public name: string,
        public state: OfficeActorChildState,
    ) {}
}

export class OfficeActorState {
    public actors: ActorInfo [];

    constructor(
        public startedAt: Date,

        public nrActorsLoaded: number,
        public lastActorLoadedAt: Date,
        public nrActorsRemoved: number,
        public lastActorRemovedAt: Date,

        public nrActorChecks: number,
        public lastActorCheckAt: Date,

        public childActorStates: any,

        public nrActorsMissed: number,
        public nrCommandsForwarded: number,
        public nrUnhandledMessages: number,
        public lastCommandForwardedAt: Date,
    ) {
        this.actors = [];

        for (let name in childActorStates) {
            this.actors.push(new ActorInfo(name, childActorStates[name]));
        }
    }
}